File: LeetCode/Python/1768. Merge Strings Alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merged_string = []
        min_length = min(len(word1), len(word2))
        
        for i in range(min_length):
            merged_string += [word1[i], word2[i]]
        
        return "".join(merged_string) + word1[min_length:] + word2[min_length:]

File: LeetCode/Java/1768. Merge Strings Alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder mergedString = new StringBuilder();
        int minLength = Math.min(word1.length(), word2.length());
        int i = 0;
        
        while (i < minLength) {
            mergedString.append(word1.charAt(i));
            mergedString.append(word2.charAt(i));
            i++;
        }
        
        mergedString.append(word1.substring(i));
        mergedString.append(word2.substring(i));
        
        return mergedString.toString();
    }
}
